feat(apigatewayv2): vpc link and private integrations (#10531)
aws/aws-cdk: packages/@aws-cdk/aws-apigatewayv2/README.md

  - [Publishing HTTP APIs](#publishing-http-apis)
  - [Custom Domain](#custom-domain)
  - [Metrics](#metrics)

>   - [VPC Link](#vpc-link)

Reviewer: Add a section after vpc link called 'private integration', mention an introduction sentence on it and then point folks at the new module.

| api | $default  |   `https://${domainName}/foo`  |
| api | beta  |   `https://${domainName}/bar`  |
| apiDemo | $default  |   `https://${domainName}/demo`  |

### Private Integration

> The `VpcLink` and private integration constructs can be used for integration of `HTTP` APIs with resource behind an Amazon VPC.
> 
> #### VPC Link
> 
> The following code uses `VpcLink` to connect an HTTP API endpoint to resources in a VPC - 

Reviewer: ```suggestion
Private integrations let HTTP APIs connect with AWS resources that are placed behind a VPC. These are usually Application
Load Balancers, Network Load Balancers or a Cloud Map service. The `VpcLink` construct enables
this integration.
The following code creates a `VpcLink` to a private VPC.
```

```ts
const vpc = new ec2.Vpc(stack, 'VPC');
const vpcLink = new VpcLink(stack, 'VpcLink', { vpc });
```

Any existing `VpcLink` resource can be imported into the CDK app via the `VpcLink.fromVpcLinkId()`.

```ts

> const stack = new Stack(app, 'my-stack');
> 

Reviewer: ```suggestion
```

const awesomeLink = VpcLink.fromVpcLinkId(stack, 'awesome-vpc-link', 'us-east-1_oiuR12Abd');
```

#### ApplicationLoadBalancerPrivateIntegration

> The following code is a basic application load balancer private integration of HTTP API:

Reviewer: This and the following sections should go into the readme for the new module `aws-apigatewayv2-integrations`.

Author: Done!

```ts
const vpc = new ec2.Vpc(stack, 'VPC');
const vpcLink = new VpcLink(stack, 'VpcLink', { vpc });
const lb = new elbv2.ApplicationLoadBalancer(stack, 'lb', { vpc });
const listener = lb.addListener('listener', { port: 80 });
listener.addTargets('target', {
  port: 80,
});

const httpEndpoint = new HttpApi(stack, 'HttpProxyPrivateApi', {
  defaultIntegration: new ApplicationLoadBalancerPrivateIntegration({
    vpcLink,
    listener,
  }),
});
```

#### NetworkLoadBalancerPrivateIntegration

The following code is a basic network load balancer private integration of HTTP API:

```ts
const vpc = new ec2.Vpc(stack, 'VPC');
const vpcLink = new VpcLink(stack, 'VpcLink', { vpc });
const lb = new elbv2.NetworkLoadBalancer(stack, 'lb', { vpc });
const listener = lb.addListener('listener', { port: 80 });
listener.addTargets('target', {
  port: 80,
});

const httpEndpoint = new HttpApi(stack, 'HttpProxyPrivateApi', {
  defaultIntegration: new NetworkLoadBalancerPrivateIntegration({
    vpcLink,
    listener,
  }),
});
```

> #### ServiceDiscoveryPrivateIntegration

Reviewer: ```suggestion
#### Service Discovery
```
